### DISCUSSION TITLE
Local dev, "port" ignored when default_short_url created in DB

### DISCUSSION THREAD

**Discussion by xazos79** [ash-jc-allen/short-url#211] [Q&A] [answered]
Hi all, firstly, love this package!

As the title suggests. When running in local dev, the default ":8000" port isn't incldued in the genereated short URL. Even if
APP_URL=http://localhost:8000

in in my ENV file.

What would be the recommended approach here?

Thanks!

**Comment by ash-jc-allen** [comment]
Hey @xazos79! Sorry you're having an issue with the package.

Would you be able to show me what values your have set in your `config/short-url.php` file, please? 😄

**Reply by xazos79** [reply]
Thanks Ash, see below:
``<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Custom Prefix
    |--------------------------------------------------------------------------
    |
    | This configuration value is used to determine the prefix that
    | is registered for the short URL route.
    |
    */
    'prefix' => '/s',

    /*
    |--------------------------------------------------------------------------
    | Middleware
    |--------------------------------------------------------------------------
    |
    | Define any middleware that the default short URL route will use.
    |
    */
    'middleware' => [
        //
    ],

    /*
    |--------------------------------------------------------------------------
    | Eloquent Factories
    |--------------------------------------------------------------------------
    |
    | Define eloquent factories that you will use for your testing purposes.
    |
    */
    'factories' => [
        \AshAllenDesign\ShortURL\Models\ShortURL::class => \AshAllenDesign\ShortURL\Models\Factories\ShortURLFactory::class,
        \AshAllenDesign\ShortURL\Models\ShortURLVisit::class => \AshAllenDesign\ShortURL\Models\Factories\ShortURLVisitFactory::class,
    ],

    /*
    |--------------------------------------------------------------------------
    | Custom Routing
    |--------------------------------------------------------------------------
    |
    | If the default route is disabled, this prevents any visitors
    | from using the default route that comes with this library.
    | This allows you to define your own route that points to
    | the controller at:
    |
    | AshAllenDesign\ShortURL\Controllers\ShortURLController
    |
    */
    'disable_default_route' => true,

    /*
    |--------------------------------------------------------------------------
    | Default URL
    |--------------------------------------------------------------------------
    |
    | Here you can override the default application base URL used to generate
    | the default short URL (default_short_url). To use your application's
    | "app.url" config value, set this field to null.
    |
    */
    'default_url' => null,

    /*
    |--------------------------------------------------------------------------
    | Forwards query parameters
    |--------------------------------------------------------------------------
    |
    | Here you can specify if the newly created short URLs will forward
    | the query parameters to the destination by default. This option
    | can be overridden when creating the short URL with the
    | ->forwardQueryParams() method.
    |
    | eg: https://yoursite.com/short/xxx?a=b => https://destination.com/page?a=b
    |
    */
    'forward_query_params' => false,

    /*
    |--------------------------------------------------------------------------
    | Enforce HTTPS in the destination URL
    |--------------------------------------------------------------------------
    |
    | Here you may specify if the visitor is redirected to the HTTPS
    | version of the destination URL by default. This option can be
    | overridden when creating the short URL with the ->secure()
    | method.
    |
    */
    'enforce_https'         => true,

    /*
    |--------------------------------------------------------------------------
    | URL Length
    |--------------------------------------------------------------------------
    |
    | The character length of the shortened URL. The 'key_length' must be
    | at least 3. However, for performance reasons, it is recommended to
    | not use a 'key_length' lower than 5.
    |
    | e.g. - Using a length of 3 would result in yourdomain.com/XXX
    |      - Using a length of 5 would result in yourdomain.com/XXXXX
    |
    | Note: This is the desired length and will act as a minimum
    |       length, not as a fixed length. For example, if all
    |       of the possible 3 character-length keys have been
    |       used, a 4 character long key will be created.
    |
    */
    'key_length'            => 6,

    /*
    |--------------------------------------------------------------------------
    | Key Salt
    |--------------------------------------------------------------------------
    |
    | Define the salt that is used to create the unique short
    | URL keys. This is used to ensure that the randomly
    | generated keys are unique.
    |
    */
    'key_salt'              => 'AshAllenDesign\ShortURL',

    /*
    |--------------------------------------------------------------------------
    | Alphabet
    |--------------------------------------------------------------------------
    |
    | Define the characters allowed in the output short URL keys.
    | The 'alphabet' must be at least 16 unique characters
    | and cannot contain spaces.
    |
    */
    'alphabet'              => 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890',

    /*
    |--------------------------------------------------------------------------
    | Tracking
    |--------------------------------------------------------------------------
    |
    | Define which fields are recorded if a shortened URL has
    | tracking enabled. Also define whether if tracking
    | is enabled by default. Each of these options can
    | be overridden when creating a short URL.
    |
    */
    'tracking'              => [
        'default_enabled' => true,

        'fields' => [
            'ip_address'               => true,
            'operating_system'         => true,
            'operating_system_version' => true,
            'browser'                  => true,
            'browser_version'          => true,
            'referer_url'              => true,
            'device_type'              => true,
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Config Validation
    |--------------------------------------------------------------------------
    |
    | Choose whether if you want the config to be validated. This
    | can be useful for ensuring that your config values are
    | safe to use.
    |
    */
    'validate_config' => false,
];

**Reply by ash-jc-allen** [reply]
Thanks for sharing this! I can't seem to reproduce this issue locally, and it looks like your config all looks okay too.

Would you please be able to confirm that you're still running into this issue after running `php artisan config:clear`? If you're still experiencing the problem, please could you set up a fresh Laravel app that has the issue so I can investigate it further? 🙂

**Comment by xazos79** [comment]
Sorry for the late reply Ash, i didn't realise you had responded.

Yep, i have run artisan config:clear and get the same results. I'm running
on wsl: ubuntu-22.04, Apache, Laravel 10, and the latest package of
short-url if that helps.

Thanks for your time!

**Reply by ash-jc-allen** [reply]
Hey! No need to apologise at all 😄

Would it be possible for you to set up a fresh Laravel app with the package that shows this issue, please? I've tried a couple of times to reproduce but can't seem to be able to. I'm not sure if I'm missing something on my end, so it'd be helpful to see a demo repo

**Comment by xazos79** [comment]
I'll try and have a crack at this at some stage. Thanks again.

**Comment by xazos79** [comment, marked as answer]
Ash, i'm really sorry.

I just realied that in my .env.local file, the APP_URL was missing the
":8000".

My .env file was correct, but the .local is being picked up in my dev
environment.

All good now. Thanks for your time.

**Reply by ash-jc-allen** [reply]
Hey! It's no problem at all. I'm really glad you managed to get it working 😄
